### ISSUE TITLE
Consider moving all library code into `delaunay.h` to make a single header library

### ISSUE THREAD

**Issue by abetusk** [Bl4ckb0ne/delaunay-triangulation#16]
Considering the size of `vector2.h`, `numeric.h`, `triangle.h` and `edge.h`, these could fit into `delaunay.h` easily and would allow people to easily add `delaunay.h` to their projects without dragging four other header files with it.

**Comment by jianyancheng** [comment]
OK,I see.Thank you very much!


 
---Original---

**Comment by Bl4ckb0ne** [comment]
I agree, the code is long due for maintenance, design wise and performance wise. I'll try to find some time in a near future to take care about that. If it's too urgent, I will gladly review any PR.

**Comment by abetusk** [comment]
I'm happy to submit a PR if you don't get around to it. I was mostly making sure this is work that you'd be open to instead of throwing a random PR at you.

**Comment by Bl4ckb0ne** [comment]
I started to think about a new design for the code. This will be my first priority in the next few weeks.
I'll make issues to tonight to guide our work, I'd like to change more than a few things :D

**Comment by chetgnegy** [comment]
I commented in #34 with some code that would make short work of this issue, too.

The functionality is contained in 200 lines, but it would take a small amount of plumbing to have this not be a breaking change.
https://pastebin.com/NtVzLSCn 

All the best!
